### PR TITLE
Topical events about page to gds

### PR DESCRIPTION
**Pull request by farahTW**
This PR transits topical events about page to GDS.
It does the following

- Duplicated topical events about page controller and show page to legacy.
- Added new show page with design system.
- Updated topical events details page with back link and caption.
- Added helper method for topical events nav items.

**Screen shots:**
image

image
Screenshot 2023-06-08 at 12 40 59

**Trello:**
https://trello.com/c/jnDUJnRu/153-about-page

### CODE REVIEW

**Comment by nnagewad** [comment]
Looks good from screenshots.

File: test/functional/admin/legacy_topical_event_about_pages_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require "test_helper"
+
+class Admin::TopicalEventAboutPagesControllerTest < ActionController::TestCase
```
[the remark this answers is not in the document]
Comment (farahTW) [reply test/functional/admin/legacy_topical_event_about_pages_controller_test.rb:3]: yeah my bad I missed it

**Comment by davidgisbey** [comment]
https://github.com/alphagov/whitehall/pull/7784/commits/47e20d0f9bc5905bcdb20a81632fe307217044df is quite a noisy commit as there's lot of changes mixed in with adding the `topical_events_nav_items` method. Would it be possible to merge all the non-related changes (backlinks, indentation etc.) into the previous commit that adds the page please.

Let me know if you need a hand with that 👍

File: test/functional/admin/topical_event_about_pages_controller_test.rb [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
-    login_as :user
     @topical_event = create(:topical_event)
+    login_as_preview_design_system_user(:writer)
   end
 
   view_test "GET show prompts user to create an about page" do
     get :show, params: { topical_event_id: @topical_event.to_param }
     assert_response :success
     assert_select "h1", @topical_event.name
-    assert_select "p", /doesn’t yet have a page/
```
Comment (davidgisbey) [inline test/functional/admin/topical_event_about_pages_controller_test.rb:13]: I think we should still be checking that the inset text renders here 

```
assert_select ".govuk-inset-text", "There is no about page associated with this topical event."

```

Comment (davidgisbey) [reply test/functional/admin/topical_event_about_pages_controller_test.rb:13]: thanks!

**Review by davidgisbey — APPROVED** [review]
lgtm!

**Pull request merged**